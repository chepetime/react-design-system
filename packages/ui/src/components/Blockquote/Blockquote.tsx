import React from "react";
import classnames from "classnames";
import css from "./Blockquote.module.scss";

export interface BlockquoteProps extends React.ComponentPropsWithRef<"div"> {
  children?: React.ReactNode;
}

export const Blockquote = React.forwardRef<HTMLDivElement, BlockquoteProps>(
  (props: BlockquoteProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.Blockquote)} ref={ref}>
        <p>Blockquote component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

Blockquote.displayName = "Blockquote";
