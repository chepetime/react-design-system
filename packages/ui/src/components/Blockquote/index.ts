export { Blockquote } from "./Blockquote";
export type { BlockquoteProps } from "./Blockquote";