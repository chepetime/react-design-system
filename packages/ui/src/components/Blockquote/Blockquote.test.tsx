
import React from "react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import { Blockquote } from "./Blockquote";
import { render, screen } from "@testing-library/react";

describe("Blockquote", () => {
  it("renders without errors", () => {
    render(<Blockquote />);
    // Assert that the component renders without throwing any errors
  });

  it("renders with children", () => {
    render(<Blockquote>Some children</Blockquote>);
    // Assert that the component renders the provided children
    const childrenElement = screen.getByText("Some children");
    expect(childrenElement).toBeInTheDocument();
  });

  it("handles user interactions", () => {
    render(<Blockquote>Some children</Blockquote>);
    // Simulate user interactions with the component
    userEvent.click(screen.getByText("Some children"));
    // Assert the expected outcome of the user interaction
    // For example, check if a certain state has changed or an event handler has been called
  });

  it("matches snapshot", () => {
    const { container } = render(<Blockquote />);
    // Assert that the rendered component matches a previously captured snapshot
    expect(container).toMatchSnapshot();
  });
});
