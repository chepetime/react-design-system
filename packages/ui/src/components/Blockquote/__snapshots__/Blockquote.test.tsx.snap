// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockquote matches snapshot 1`] = `
<div>
  <div
    class="Blockquote"
  >
    <p>
      Blockquote component working!
    </p>
    <div />
  </div>
</div>
`;
